Tools/travis_compile.sh

```diff
@@ -3,6 +3,27 @@ set -xe
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
+mkdir -p sdkcache/
+git submodule | cut -d'(' -f1 | sort > sdkcache/current.txt
```

Comment:
We are only concerned with the SHA1 of the actual submodules in the repo, not the submodules in the submodules. The reasoning is that if you update a submodule of a submodule, when you go to git-add/git-commit it, that will result in a new SHA1 that must in turn be added/committed to the repo, which will show as a different SHA1... meaning, recursion is irrelevant for the purposes of knowing when submodules have changed.